File: PlaneWar/Client_1/Assets/Scripts/Libs/FrameHandle/TickObjectNoPriorityMgr.cs
namespace SDK.Lib
{
    // 每一帧执行的对象管理器
    public class TickObjectNoPriorityMgr : DelayNoPriorityHandleMgr, ITickedObject, IDelayHandleItem, INoOrPriorityObject
    {
        public TickObjectNoPriorityMgr()
        {

        }

        override public void init()
        {
            base.init();
        }

        override public void dispose()
        {
            base.dispose();
        }

        public void setClientDispose(bool isDispose)
        {

        }

        public bool isClientDispose()
        {
            return false;
        }

        public void onTick(float delta)
        {
            this.incDepth();

            this.onPreAdvance(delta);
            this.onExecAdvance(delta);
            this.onPostAdvance(delta);

            this.decDepth();
        }

        virtual protected void onPreAdvance(float delta)
        {

        }

        virtual protected void onExecAdvance(float delta)
        {
            int idx = 0;
            int count = this.mNoPriorityList.Count();
            ITickedObject tickObject = null;

            while (idx < count)
            {
                tickObject = this.mNoPriorityList.get(idx) as ITickedObject;

                if (null != (tickObject as IDelayHandleItem))
                {
                    if (!(tickObject as IDelayHandleItem).isClientDispose())
                    {
                        tickObject.onTick(delta);
                    }
                }
                else
                {
                    if (MacroDef.ENABLE_LOG)
                    {
                        Ctx.mInstance.mLogSys.log("TickObjectNoPriorityMgr::onExecAdvance, failed", LogTypeId.eLogCommon);
                    }
                }

                ++idx;
            }
        }

        virtual protected void onPostAdvance(float delta)
        {

        }
    }
}

File: PlaneWar/Client_1/Assets/LuaFramework/Lua/MyLua/Libs/FrameHandle/TickObjectNoPriorityMgr.lua
MLoader("MyLua.Libs.Core.GlobalNS");
MLoader("MyLua.Libs.Core.Class");
MLoader("MyLua.Libs.Core.GObject");

MLoader("MyLua.Libs.DelayHandle.DelayNoPriorityHandleMgr");

-- 每一帧执行的对象管理器
local M = GlobalNS.Class(GlobalNS.DelayNoPriorityHandleMgr);
M.clsName = "TickObjectNoPriorityMgr";
GlobalNS[M.clsName] = M;

function M:ctor()

end

function M:init()
	M.super.init(self);
end

function M:dispose()
	M.super.dispose(self);
end

function M:setClientDispose(isDispose)
	
end

function M:isClientDispose()
	return false;
end

function M:onTick(delta, tickMode)
	self:incDepth();

	self:onPreAdvance(delta, tickMode);
	self:onExecAdvance(delta, tickMode);
	self:onPostAdvance(delta, tickMode);

	self:decDepth();
end

function M:onPreAdvance(delta, tickMode)

end

function M:onExecAdvance(delta, tickMode)
	local idx = 0;
	local count = self.mNoOrPriorityList:Count();
	local tickObject = nil;

	while (idx < count) do
		tickObject = self.mNoOrPriorityList:get(idx);

		if (nil ~= tickObject) then
			if (not tickObject:isClientDispose()) then
				tickObject:onTick(delta, tickMode);
			end
		else
			if (MacroDef.ENABLE_LOG) then
				GCtx.mLogSys:log("TickObjectNoPriorityMgr::onExecAdvance, failed", GlobalNS.LogTypeId.eLogCommon);
			end
		end

		idx = idx + 1;
	end
end

function M:onPostAdvance(delta, tickMode)

end

return M;